src/Mado/QueryBundle/Repositories/HttpRepository.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Mado\QueryBundle\Repositories;
+
+use Mado\QueryBundle\Repositories\BaseRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class HttpRepository
+{
+    private $manager;
+
+    private $stack;
+
+    private $repo;
+
+    public function __construct(
+        EntityManagerInterface $manager,
+        RequestStack $stack
+    ) {
+        $this->manager = $manager;
+        $this->stack = $stack;
+    }
+
+    public function setRepo(BaseRepository $repo)
+    {
+        $this->repo = $repo;
+    }
+
+    public function buildForEntity(string $entityClassName)
+    {
+        if (!$this->repo) {
+            $this->repo = new BaseRepository(
+                $this->manager,
+                $this->manager->getClassMetadata($entityClassName)
+            );
+        }
+
+        $this->repo->setRequest($this->stack->getCurrentRequest());
```

Comment:
@orociscaverai I started to play with QueryBundle few days ago and the result is here. A way to hide `setRequest`.

````php
    $httpRepository->buildForEntity(\App\Entity\Initiatives::class);
    $httpRepository->getResult();
```

May we work together with #170 and #167?

Reply:
@orociscaverai what do you think?

Reply:
up

Reply:
OK